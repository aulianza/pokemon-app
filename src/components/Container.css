.container {
    max-width: 900px;
    margin: auto;
    padding: 2rem 1rem 5rem;
    display: flex;
    flex-direction: column;
    min-height: calc(100vh - 10rem);
}