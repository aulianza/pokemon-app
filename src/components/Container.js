import React from "react";
import styled from "@emotion/styled";

const Container = (props) => {
    return <StyledContainer>{props.children}</StyledContainer>;
};

export default Container;

const StyledContainer = styled.div`
    max-width: 900px;
    margin: auto;
    padding: 2rem 1rem 5rem;
    display: flex;
    flex-direction: column;
    min-height: calc(100vh - 10rem);
`;
