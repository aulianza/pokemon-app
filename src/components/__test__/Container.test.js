import Container from "../Container";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { BrowserRouter as Router } from "react-router-dom";

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("container-component, snapshot", () => {
    act(() => {
        render(
            <Router>
                <Container />
            </Router>,
            container
        );
    });
    expect(container).toMatchSnapshot();
});
