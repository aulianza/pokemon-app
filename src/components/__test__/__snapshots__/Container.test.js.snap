// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container-component, snapshot 1`] = `
<div>
  <div
    class="container"
  />
</div>
`;
